corehq/apps/app_manager/models.py

```diff
@@ -5755,6 +5769,23 @@ def for_app(cls, app):
 
     def save(self, *args, **kwargs):
         LatestAppInfo(self.app_id, self.domain).clear_caches()
+
+        # Save to SQL
+        model, created = SQLGlobalAppConfig.objects.get_or_create(
+            domain=self.domain,
+            app_id=self.app_id,
+            defaults={
+                'apk_version': LATEST_APK_VALUE,
+                'app_version': LATEST_APP_VALUE,
+            }
+        )
+        model.app_prompt = self.app_prompt
+        model.apk_prompt = self.apk_prompt
+        model.app_version = self.app_version
+        model.apk_version = self.apk_version
+        model.save()
```

Comment:
Two writes (create + save) is brutal. Could save a write by doing `SQLGlobalAppConfig.objects.get(...)` and if it does not exist then create a new record else update the existing record (if necessary). Theoretically it would be possible to do it all with a single UPSERT query (rather than get + insert or update), but I'm not sure if Django makes that easy.

Reply:
[`update_or_create()`](https://docs.djangoproject.com/en/1.11/ref/models/querysets/#update-or-create) might be what you're looking for.

Reply:
d4de3c6fb8c

Reply:
d4de3c6fb8c